builtin/providers/digitalocean/resource_digitalocean_loadbalancer.go

```diff
@@ -0,0 +1,292 @@
+				ForceNew: true,
+			},
+
+			"algorithm": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "round_robin",
+			},
+
+			"forwarding_rule": {
+				Type:     schema.TypeList,
+				Required: true,
+				MinItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"entry_protocol": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"entry_port": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"target_protocol": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"target_port": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"certificate_id": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"tls_passthrough": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  false,
+						},
+					},
+				},
+			},
+
+			"healthcheck": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"protocol": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"port": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"path": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"check_interval_seconds": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Default:  10,
+						},
+						"response_timeout_seconds": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Default:  5,
+						},
+						"unhealthy_threshold": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Default:  3,
+						},
+						"healthy_threshold": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Default:  5,
+						},
+					},
+				},
+			},
+
+			"sticky_sessions": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "none",
+						},
+						"cookie_name": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"cookie_ttl_seconds": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+
+			"droplet_ids": {
+				Type:     schema.TypeList,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Optional: true,
+			},
+
+			"droplet_tag": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"redirect_http_to_https": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"ip": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func buildLoadBalancerRequest(d *schema.ResourceData) (*godo.LoadBalancerRequest, error) {
+	opts := &godo.LoadBalancerRequest{
+		Name:                d.Get("name").(string),
```

Comment:
I ask mostly because the included tests all include a `name` attribute, and I noticed it's optional. Also because I don't know DO's API so I just wanted to be safe / learn